src/http.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::{self, Write};
```

Comment:
I think it ok, [doc](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/writing_clientlibs.md#writing-client-libraries) encourages making client libraries easy to use.